src/Exception/Failure/FailureConverter.php

```diff
@@ -206,39 +205,36 @@ private static function createFailureException(Failure $failure, DataConverterIn
                     $previous
                 );
 
-            case $failure->hasTimeoutFailureInfo():
+            case $failure->getTimeoutFailureInfo() !== null:
                 $info = $failure->getTimeoutFailureInfo();
 
-                $details = $info->hasLastHeartbeatDetails()
-                    ? EncodedValues::fromPayloads($info->getLastHeartbeatDetails(), $converter)
-                    : EncodedValues::empty()
-                ;
+                $details = $info->getLastHeartbeatDetails() === null
+                    ? EncodedValues::empty()
+                    : EncodedValues::fromPayloads($info->getLastHeartbeatDetails(), $converter);
 
                 return new TimeoutFailure($failure->getMessage(), $details, $info->getTimeoutType(), $previous);
 
-            case $failure->hasCanceledFailureInfo():
+            case $failure->getCanceledFailureInfo() !== null:
                 $info = $failure->getCanceledFailureInfo();
 
-                $details = $info->hasDetails()
-                    ? EncodedValues::fromPayloads($info->getDetails(), $converter)
-                    : EncodedValues::empty()
-                ;
+                $details = $info->getDetails() === null
+                    ? EncodedValues::empty()
+                    : EncodedValues::fromPayloads($info->getDetails(), $converter);
 
                 return new CanceledFailure($failure->getMessage(), $details, $previous);
 
-            case $failure->hasTerminatedFailureInfo():
+            case $failure->getTerminatedFailureInfo() !== null:
                 return new TerminatedFailure($failure->getMessage(), $previous);
 
-            case $failure->hasServerFailureInfo():
+            case $failure->getServerFailureInfo() !== null:
```

Comment:
There are a lot of changes like this.
That related with the fact I used an old Protoc version and it removed all `has*` methods after compilation.
I think I should revert it.